### PR TITLE
DynamoDB: Add special decoding for varied lists

### PR DESCRIPTION
**Pull request by amotl**
## About
Because CrateDB can't store varied lists into `OBJECT(DYNAMIC)` columns, let's break them apart and store them into a separate `OBJECT(IGNORED)` column.

## References
- GH-28

/cc @matriv, @proddata

### CODE REVIEW

File: src/commons_codec/transform/dynamodb.py [hunk 1]
```diff
@@ -37,14 +39,45 @@ def _deserialize_ss(self, value):
...
 
-    # Define name of the column where CDC's record data will get materialized into.
-    DATA_COLUMN = "data"
+    # Define name of the column where typed DynamoDB fields will get materialized into.
+    # This column uses the `OBJECT(DYNAMIC)` data type.
+    TYPED_COLUMN = "data"
+
+    # Define name of the column where untyped DynamoDB fields will get materialized into.
+    # This column uses the `OBJECT(IGNORED)` data type.
+    UNTYPED_COLUMN = "aux"
```
Comment (amotl) [inline src/commons_codec/transform/dynamodb.py:80]: Running out of "naming things", please advise if you can come up with better labels.

File: tests/transform/test_dynamodb_types_cratedb.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            Decimal("1"),
+            "foo",
+            [Decimal("1.25")],
+        ]
+
+
+def test_decode_typed_untyped():
+    assert DynamoDBCDCTranslator(table_name="foo").decode_record(
+        {"foo": {"N": "84.84"}, "bar": {"L": [{"N": "1"}, {"S": "foo"}]}}
+    ) == DualRecord(typed={"foo": 84.84}, untyped={"bar": [1.0, "foo"]})
```
Comment (amotl) [inline tests/transform/test_dynamodb_types_cratedb.py:23]: The gist, in outcome, is best viewed here.

File: tests/transform/test_dynamodb_full.py [hunk 3]
```diff
@@ -71,6 +72,7 @@ def test_to_sql_list_of_objects():
                         "utm_source": "google",
                     }
                 ]
-            }
+            },
+            "untyped": {},
```
Comment (seut) [inline tests/transform/test_dynamodb_full.py:76]: Shouldn't we add a test including `untyped` columns?

Comment (amotl) [reply tests/transform/test_dynamodb_full.py:76]: You are right on the spot. Another test for that is yet missing. Thanks a stack!

File: tests/transform/test_dynamodb_full.py [hunk 4]
```diff
@@ -35,32 +36,40 @@
...
+            },
+            "untyped": {
+                "list_varied": [
+                    {"a": 1.0},
+                    2.0,
+                    "Three",
+                ],
+            },
         },
     )
```
Comment (amotl) [inline tests/transform/test_dynamodb_full.py:65]: Missing test case fragment has been added. /cc @seut

**Pull request merged**